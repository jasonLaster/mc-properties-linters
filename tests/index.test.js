const fs = require("fs");
const path = require("path");
const { parse } = require("../src/parse");

const { hasLabel } = require("../src/tests");

function getSource(name) {
  return fs.readFileSync(path.join(__dirname, `./fixtures/${name}`), "utf8");
}

describe("parsing", () => {
  it("simple", () => {
    const text = getSource("simple.txt");
    const data = parse(text);
    expect(data).toMatchSnapshot();
  });

  it("bad", () => {
    const text = getSource("bad.txt");
    const data = parse(text);
    expect(data).toMatchSnapshot();
  });
});
