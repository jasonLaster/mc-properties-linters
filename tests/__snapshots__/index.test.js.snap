// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parsing bad 1`] = `
Array [
  Object {
    "comments": Array [
      "# LOCALIZATION NOTE copySource): This is the text that appears in the",
      "# context menu to copy the selected source of file open.",
    ],
    "l10n": Array [
      Object {
        "key": "copySource",
        "value": "Copy",
      },
      Object {
        "key": "copySource.accesskey",
        "value": "y",
      },
    ],
  },
  Object {
    "comments": Array [
      "# LOCALIZATION NOTE (copySource): This is the text that appears in the",
      "# context menu to copy the selected source of file open.",
    ],
    "l10n": Array [
      Object {
        "key": "copySorce",
        "value": "Copy",
      },
    ],
  },
  Object {
    "comments": Array [
      "# LOCALIZATION NOTE (copySource): This is the text that appears in the",
      "# context menu to copy the selected source of file open.",
    ],
    "l10n": Array [
      Object {
        "key": "copySource",
        "value": "Copy",
      },
      Object {
        "key": "copySorce.accesskey",
        "value": "y",
      },
    ],
  },
]
`;

exports[`parsing simple 1`] = `
Array [
  Object {
    "comments": Array [
      "# LOCALIZATION NOTE (collapsePanes): This is the tooltip for the button",
      "# that collapses the left and right panes in the debugger UI.",
    ],
    "l10n": Array [
      Object {
        "key": "collapsePanes",
        "value": "Collapse panes",
      },
    ],
  },
  Object {
    "comments": Array [
      "# LOCALIZATION NOTE (copySource): This is the text that appears in the",
      "# context menu to copy the selected source of file open.",
    ],
    "l10n": Array [
      Object {
        "key": "copySource",
        "value": "Copy",
      },
      Object {
        "key": "copySource.accesskey",
        "value": "y",
      },
    ],
  },
]
`;
